### ISSUE TITLE
Add math funcs as methods to AffineScalarFunc

### ISSUE THREAD

**Issue by pkienzle** [lebigot/uncertainties#130]
The numpy math ufuncs such as cos() have a feature that check if an object has the cos() method and call that. Therefore, if you add math functions to AffineScalarFunc class then uncertainty values can be used interchangeably with python floats when called with numpy.cos.

For example:
```
import numpy as np
from uncertainties import ufloat, umath, core
core.AffineScalarFunc.cos = umath.cos
print(ufloat(1, 0.1).cos())
print(np.cos(ufloat(1, 0.1))
```
produces:
```
0.54+/-0.08
0.54+/-0.08
```

This allows existing code written with numpy math functions to work with uncertain values without change.

**Comment by lebigot** [comment]
This is indeed a change that makes sense. It was mentioned in issue #47, so I'll close this issue. If my memory serves, at the time the NumPy mechanism was quite new, but now it is probably very well established: this is a good reason to implement it.

Now, I was under the impression that supporting NumPy operations transparently required creating a special class for NumPy arrays with uncertainties, but it looks like the situation is simpler than this. 

In conclusion, I think that this is a worthwhile improvement to the uncertainties package. I must say that I don't have much time to invest in this library, so it would be nice if someone can pick this up (and produce clear and clean code and documentation). I'll do my best, though, because this would make the life of users a bit easier.
